### PR TITLE
add default_on parameter on PolymorficList and PolymorphicObject

### PR DESCRIPTION
**Pull request by bjamet**

### CODE REVIEW

File: lupin/fields/polymorphic_list.py [hunk 1]
```diff
@@ -16,15 +16,18 @@ class PolymorphicList(Field):
                             "painting": painting_schema
                         })
     """
-    def __init__(self, on, schemas, **kwargs):
+    def __init__(self, on, schemas, default_on=None, **kwargs):
```
Comment (holinnn) [inline lupin/fields/polymorphic_list.py]:
```suggestion
    def __init__(self, on, schemas, default_type=None, **kwargs):
```

File: lupin/fields/polymorphic_list.py [hunk 2]
```diff
@@ -16,15 +16,18 @@ class PolymorphicList(Field):
...
         """
         Args:
             on (str): JSON key used to get the object type
             schemas (dict): schemas used for each values used for the `on` key
+            default_on (str): default value if the 'on' field is not available
```
Comment (holinnn) [inline lupin/fields/polymorphic_list.py]:
```suggestion
            default_type (str): default value if the 'on' field is not available
```

File: lupin/fields/polymorphic_list.py [hunk 3]
```diff
@@ -16,15 +16,18 @@ class PolymorphicList(Field):
...
+              usefull for update of schema from List to PolymorphicList
         """
         merge_validator(kwargs, Type(list))
         super(PolymorphicList, self).__init__(**kwargs)
         self._on = on
+        self._default_on = default_on
```
Comment (holinnn) [inline lupin/fields/polymorphic_list.py]:
```suggestion
        self._default_type = default_type
```

File: lupin/fields/polymorphic_list.py [hunk 4]
```diff
@@ -41,7 +44,7 @@ def load(self, value, mapper):
         if value is None:
             return None
 
-        return [mapper.load(item, self._schemas_by_json_value[item[self._on]])
+        return [mapper.load(item, self._schemas_by_json_value[item.get(self._on, self._default_on)])
```
Comment (holinnn) [inline lupin/fields/polymorphic_list.py]:
```suggestion
        return [mapper.load(item, self._schemas_by_json_value[item.get(self._on, self._default_type)])
```

File: lupin/fields/polymorphic_object.py [hunk 5]
```diff
@@ -14,15 +14,19 @@ class PolymorphicObject(Field):
                               "painting": painting_schema
                           })
     """
-    def __init__(self, on, schemas, **kwargs):
+    def __init__(self, on, schemas, default_on = None, **kwargs):
         """
         Args:
             on (str): JSON key used to get the object type
             schemas (dict): schemas used for each values used for the `on` key
+            default_on (str): default value if the 'on' field is not available
```
Comment (holinnn) [inline lupin/fields/polymorphic_object.py]:
```suggestion
            default_type (str): default value if the 'on' field is not available
```

File: lupin/fields/polymorphic_object.py [hunk 6]
```diff
@@ -14,15 +14,19 @@ class PolymorphicObject(Field):
...
+              usefull for update of schema from List to PolymorphicList
+
         """
         merge_validator(kwargs, Type(dict))
         super(PolymorphicObject, self).__init__(**kwargs)
         self._on = on
+        self._default_on = default_on
```
Comment (holinnn) [inline lupin/fields/polymorphic_object.py]:
```suggestion
        self._default_type = default_type
```

File: lupin/fields/polymorphic_object.py [hunk 7]
```diff
@@ -14,15 +14,19 @@ class PolymorphicObject(Field):
                               "painting": painting_schema
                           })
     """
-    def __init__(self, on, schemas, **kwargs):
+    def __init__(self, on, schemas, default_on = None, **kwargs):
```
Comment (holinnn) [inline lupin/fields/polymorphic_object.py]:
```suggestion
    def __init__(self, on, schemas, default_type = None, **kwargs):
```

File: lupin/fields/polymorphic_object.py [hunk 8]
```diff
@@ -39,7 +43,7 @@ def load(self, value, mapper):
         if value is None:
             return None
 
-        schema = self._schemas_by_json_value[value[self._on]]
+        schema = self._schemas_by_json_value[value.get(self._on, self._default_on)]
```
Comment (holinnn) [inline lupin/fields/polymorphic_object.py]:
```suggestion
        schema = self._schemas_by_json_value[value.get(self._on, self._default_type)]
```

File: tests/lupin/fields/test_polymorphic_list.py [hunk 9]
```diff
@@ -4,22 +4,25 @@
...
+    mapper.register(Money, money_schema)
     return PolymorphicList(on="type",
                            schemas={
                                "jewel": jewel_schema,
-                               "painting": painting_schema
-                           })
+                               "painting": painting_schema,
+                               "money": money_schema
+                           },
+                           default_on="money")
```
Comment (holinnn) [inline tests/lupin/fields/test_polymorphic_list.py]:
```suggestion
                           default_type="money")
```

File: tests/lupin/fields/test_polymorphic_object.py [hunk 10]
```diff
@@ -4,22 +4,25 @@
...
+    mapper.register(Money, money_schema)
     return f.PolymorphicObject(on="type",
                                schemas={
                                    "jewel": jewel_schema,
-                                   "painting": painting_schema
-                               })
+                                   "painting": painting_schema,
+                                   "money": money_schema
+                               },
+                               default_on="money")
```
Comment (holinnn) [inline tests/lupin/fields/test_polymorphic_object.py]:
```suggestion
                               default_type="money")
```

File: lupin/fields/polymorphic_list.py (hunk 1 above)
Comment (bjamet) [reply lupin/fields/polymorphic_list.py]: well, in general we use "type" as value of th "on" parameter : 
f.PolymorphicObject(on="type",...)
I think you have this usage in mind when you propose "default_type" 
but if use on other field, 'default_type' is less logical
f.PolymorphicObject(on="kind", default_type="jewel", )

Comment (holinnn) [reply lupin/fields/polymorphic_list.py]: For me the `on` parameter define the name of the field that contains the type to instantiate.
So having `on` defining the name of the field and `default_on` containing the default value of that field is kind of misleading.

**Comment by holinnn** [comment]
tack så mycket!
You can merge and publish 🙂

**Comment by bjamet** [comment]
Done, thank you for you time

**Pull request merged**